src/views/Orders.vue

```diff
@@ -586,6 +577,51 @@ export default defineComponent({
         }
       })
       return rejectOrderModal.present()
+    },
+
+    async openProofOfDeliveryModal(order: any) {
+      const modal = await modalController.create({
+        component: ProofOfDeliveryModal,
+        componentProps: {
+          order
+        },
+      });
+
+      await modal.present();
+      
+      const { data } = await modal.onDidDismiss();
+      
+      if (data?.confirmed && data?.proofOfDeliveryData) {
+        emitter.emit("presentLoader");
+        
+        try {
+          // First deliver the shipment
+          await this.deliverShipment(order);
+          
+          // Then send the proof of delivery email
+          const resp = await OrderService.sendPickupNotification(data.proofOfDeliveryData);
+          
+          if (hasError(resp)) {
+            logger.error("Pickup notification failed:", resp);
+            showToast(translate("Order delivered but failed to send handover email"));
+          } else {
+            showToast(translate("Order delivered and handover email sent successfully"));
+          }
+        } catch (err) {
+          logger.error("Error in handover process:", err);
+          showToast(translate("Something went wrong during handover"));
```

Comment:
Toast messages in this flow is misleading, as this flow only handles the case when pickup notification flow fails,  error in `deliverShipment` flow is already handled in specific action.

Also, if the delivery shipment flow fails then also the notification api will be called, so need to update the whole flow in such a way that one failure should not make furthur calls and should prompt the user accordingly.